src/sentry/api/endpoints/group_notes_details.py

```diff
@@ -52,9 +52,12 @@ def put(self, request, group, note_id):
 
         if serializer.is_valid():
             # Would be nice to have a last_modified timestamp we could bump here
+            external_id = note.data.get('external_id')
```

Comment:
Should this be part of the serializer as an optional field? We could at least constrain the types to int/str. Right now unpleasant folks could send us lists.